### DISCUSSION TITLE
High CPU during download

### DISCUSSION THREAD

**Discussion by loeffelpan** [antlafarge/jdownloader#31] [Q&A] [answered]
Can anyone else reproduce high CPU while downloading?
Never had JD2 in a docke container before, but with this image I have two cores busy with `java` on my docker host when downloading a file.

Not tested yet, but I would bet that CPU is not that high with JD2 and openjdk directly on the same host.

Any thoughts?

**Comment by antlafarge** [comment, marked as answer]
Good question, I didn't check the CPU load, compared to JD2 and OpenJDK on the same host.
By default Docker gives to the container full access to CPUs.
So I don't see why it will take more resource though Docker, I think you will have the same CPU load without Docker.
This image starts JD2 through OpenJDK exactly the same way you start it on your host machine.
See here : https://github.com/antlafarge/jdownloader/blob/50f2cefc8a6443b9f02577f9e2ffe23cb344a0e3/docker-entrypoint.sh#L113

Anyway did you try these options to limit the CPU usage of the container?
https://docs.docker.com/config/containers/resource_constraints/#cpu

And if anyone has some metrics on JD2 load in a container and on the host machine, or similar issues, please let me know :)

**Reply by loeffelpan** [reply]
Just tested with the same cfg-directory with local and dockerized container for java and JD2.
As you expected. No difference in CPU load.

Sorry for bothering you.

**Comment by antlafarge** [comment]
No problem, thank you for sharing your experience, now we know that JD2 requires a lot of CPU during the downloads and that we can customize the container options to restrict CPU resources.
https://docs.docker.com/config/containers/resource_constraints/#cpu
